packages/components/src/border-box-control/stories/index.js

```diff
@@ -84,12 +84,12 @@ Default.args = {
 		style: 'dashed',
 		width: '1px',
 	},
-	__next36pxDefaultSize: false,
+	__next40pxDefaultSize: false,
 	popoverPlacement: 'right-start',
 };
 
 const WrapperView = styled.div`
-	max-width: 280px;
+	max-width: 248px;
```

Comment:
From memory, this was to help better demonstrate the space available when used within the sidebar. The component can adjust to whatever width, though, without this tweak, you had to load the editor to see the component in the wild and constrained by our sidebar's width and panel padding.

I can happily remove this change. Personally, I'm confused as to why one arbitrary width for demonstration purposes is worse than another. At the least, this helped save time while iterating on these components so it has served its purpose 🙂

Reply:
>  Personally, I'm confused as to why one arbitrary width for demonstration purposes is worse than another.

My point here was that we should not have a `max-width` at all — a component should not be aware / be tightly coupled to the context in which it's currently used (e.g the sidebar) (cc @noisysocks who made a very similar change in #44559 (

We can definitely leave this change as-is for this PR, but we should look into removing `max-width` in the future.

> At the least, this helped save time while iterating on these components so it has served its purpose 🙂

You should be able to use Storybook for these purposes, and resize the canvas to the desired size to test the component's responsiveness

Reply:
> and likely let the parent component dictate the width (see, for example, https://github.com/WordPress/gutenberg/pull/44559)

FWIW, I think that example is a little different in that the max-width was within the component's styles. The `BorderBoxControl` does let the parent component dictate the width, which is what the `WrapperView` in the story was doing so we can see the impacts of the large input with limited width like in the sidebar.

> Storybook should offer a few tools to make it easier to resize the canvas and test the component's responsiveness

This sounds like a good follow-up, adding some tools to resize the canvas and remove the max-widths across multiple components in the storybook. I don't think it should block this PR though.

Reply:
Storybook already offers some tools to resize the canvas:

https://user-images.githubusercontent.com/1083581/196409167-be7b9a68-fd56-4ce3-b645-23f9e3a66e53.mp4


Although I agree that this is not blocking.

Reply:
Thanks. I should have been clearer. I meant tools to resize to some meaningful and useful preset sizes, e.g. our sidebar widths. The device sizes don't really help, and manually drag resizing the canvas is inaccurate, especially when trying to resolve 1 or 2 pixel spacing issues.

Personally, I find it cumbersome to have to be resizing the canvas constantly and a poorer reflection on the components when they are stretched out of realistic proportions. I understand the idea that removing the restriction illustrates how the components adapt. For me, though, I think we lose as much as we gain. I'm not trying to change our direction here just highlight some of the friction experienced when using the storybook for development, not documentation.

Reply:
Absolutely! This kind of feedback is incredibly valuable to us, so keep it coming!

As you mentioned, it's hard to strike the right balance between documenting and developing — I hope that setting the correct expectations with our users and that adopting a consistent approach throughout all examples (e.g. all components have unconstrained width?) can be a good direction to take.

(cc @mirka)

Reply:
My first choice would've been to add a "WP sidebar" preset to the [Viewport tool](https://storybook.js.org/docs/react/essentials/viewport), but that's not what we want here since changing the viewport itself would affect the font size in text fields.

So instead we'll need to add another toolbar add-on to wrap the stories in a div with some preset max-widths. This is super easy to build, so let me know if that would work for you.

Reply:
> So instead we'll need to add another toolbar add-on to wrap the stories in a div with some preset max-widths. This is super easy to build, so let me know if that would work for you.

That sounds like a good compromise. 

I took the liberty of having a go at creating the toolbar add-on ( https://github.com/WordPress/gutenberg/pull/45134 ). Let me know if I missed anything. If that PR is suitable, I'll go through our components and remove those ad hoc wrappers in a separate follow-up.

Reply:
I intended to remove all these wrappers across multiple components in a follow-up PR. Happy to quickly strip it from here now to get the ball rolling, though 🙂

Reply:
✅ Done: https://github.com/WordPress/gutenberg/pull/41860/commits/072f6afb4905784b572ab4c3e4a576bee067bb93

Reply:
Sorry, I didn't understand that was the plan! Thank you anyway!

Reply:
All good. I should have re-iterated my plans after the conversation evolved. 

I've put up a quick PR removing wrappers on non-border controls: https://github.com/WordPress/gutenberg/pull/45305.